servant/src/Servant/Utils/Links.hs

```diff
@@ -137,16 +137,19 @@ data Link = Link
   } deriving Show
 
 -- | If either a or b produce an empty constraint, produce an empty constraint.
+--
+-- This works because of:
+-- https://ghc.haskell.org/trac/ghc/wiki/NewAxioms/CoincidentOverlap
```

Comment:
Won't this show up in the haddocks? I feel like it's a good comment to have in the source code, but not relevant to someone just browsing the haddocks.